core/src/test/java/org/apache/iceberg/hadoop/HadoopFileIOTest.java

```diff
@@ -175,10 +181,16 @@ private List<Path> createRandomFiles(Path parent, int count) {
                 Path path = new Path(parent, "file-" + i);
                 paths.add(path);
                 fs.createNewFile(path);
+                createdFiles.increment();
               } catch (IOException e) {
                 throw new UncheckedIOException(e);
               }
             });
+
+    Awaitility.await()
```

Comment:
nit:
CountDownLatch could help too, but it would mean hadling InterruptedException in the method
```
    CountDownLatch latch = new CountDownLatch(count);
    random
        .ints(count)
        . parallel()
        .forEach(
            i -> {
              try {
                Path path = new Path(parent, "file-" + i);
                paths.add(path);
                fs.createNewFile(path);
                latch.countDown();
              } catch (IOException e) {
                throw new UncheckedIOException(e);
              }
            });
    latch.await();

Reply:
I'd prefer to keep the assertion check to make sure we actually created the required number of files